docs/How-to-implement-a-new-ci-test.md

```diff
@@ -0,0 +1,146 @@
+# How to extend GPUCSL by a new distribution
+
+This is a guide on how you can embed your own implementation of an CI algorithm. 
+
+Testing your code is optional, but highly recommended. This project supports multiple types of tests:
+- Python tests via pytest. They target the python code you write.
+- CUDA tests from python. They target your CUDA code callable from python (prefixed by __global__ and with return value void).
+    They enable you to prepare your data with python packages like numpy so you can make clearer what your are testing (example 
+    data + call in numpy instead of randomly, externally generated input that you hardcode)
+- Google test: Enables to test CUDA code not covered by the previous point or if you simply prefer to test your CUDA in C++
+
+
+Now following are the necessary steps for making a new distribution acessible in GPUCSL. We recommended to follow the given order of steps.
+
+
+## Implement your CUDA kernel
+
+This is the part we can help you with the least. You have to design efficient CUDA kernel code. While doing so you can use 
+any of the already existing helper functions and you can reuse code from our distributions however you like.
+
+You have to place your cuda files in the gpucsl/pc/cuda or gpucsl/pc/cuda/helpers directories! (otherwise you have to modify
+the compiler_options variable in the get_module function of gpucsl/pc/kernel_management.py like it is done for cuda_dir or cuda_helpers_dir)
+
+Note: If you want you can use templates. As the CUDA code is compiled during the runtime of GPUCSL (!) there are already some 
+runtime informations, like the number of variables, accessible. For more information see the next section.
+
+
+## Implement your own Kernel Management
+
+The Kernel Management encapsulates the details of kernel access, for example the grid and block mapping of the kernel is defined here.
+Otherwise encapsulates details of the underlying cupy libraray (which cuda file to read, it compiles the code on creation of the class, function name resulution to level). 
+
+To write your own you want inherit from the abstract base class Kernel. You then have to implement the following methods:
```

Comment:
```suggestion
The Kernel Management encapsulates the details of kernel access, for example, the grid and block mapping of the kernel is defined here.
Otherwise encapsulates details of the underlying CuPy library (which CUDA file to read, it compiles the code on the creation of the class, function name resolution to level). 

To write your own Kernel Management you want to inherit from the abstract base class Kernel. You then have to implement the following methods:
```